### ISSUE TITLE
is compile really necessary when we predict using a pretrained model?

### ISSUE THREAD

**Issue by dongzhuoyao** [fchollet/keras#3074]
I search a  result from stackoverflow:

[http://stackoverflow.com/questions/33474424/keras-load-weights-of-a-neural-network-error-when-predicting](http://stackoverflow.com/questions/33474424/keras-load-weights-of-a-neural-network-error-when-predicting)

it says you must compile before predict,but in my current keras version,I remove the compile code block,and then also get a good predict result.

from my opinion,when we predict ,we just do forward-propogation rather than back-propogation,so we dont need know what the loss is  and how the loss is comprised of.so for a better architecture design,we shouldnt need "compile".

so finally,i want to confirm  whether we need "complie" before we "predict"??yes or no

**Comment by ChristianThomae** [comment]
Since this commit https://github.com/fchollet/keras/commit/d8864bfe48d64c15dc70f13e46c7e08772811fd9 or keras version 1.0.3 it is no longer necessary to compile the model to use predict.

**Comment by rohith14** [comment]
I read that to profile keras (and TF) I need to enable trace for TF session as shown [here](https://medium.com/@xianbao.qian/find-the-bottleneck-of-your-keras-model-using-tf-trace-2a6757aa372).  But, after this (https://github.com/keras-team/keras/issues/3074), compile is not necessary for calling predict. I am not sure how to pass run_metadata options for timeline. can you please help?
